components/camera/videosource/webcam.go

```diff
@@ -413,11 +415,10 @@ func (c *webcam) Image(ctx context.Context, mimeType string, extra map[string]in
 	if c.reader == nil {
 		return nil, camera.ImageMetadata{}, errors.New("underlying reader is nil")
 	}
-	img, release, err := c.reader.Read()
-	if err != nil {
-		return nil, camera.ImageMetadata{}, err
+	img := c.buffer.GetLatestFrame()
+	if img == nil {
+		return nil, camera.ImageMetadata{}, errors.New("no frames available to read")
 	}
-	defer release()
 
 	if mimeType == "" {
 		mimeType = utils.MimeTypeJPEG
```

Comment:
2 lines down from here, the call to `rimage.EncodeImage` might be trouble: perhaps we're taking a raw array of pixels and doing an FFT on it in the process of turning it into a jpeg. This does not happen up in `Images()` above: I wonder if there's something in there that's a real bottleneck.

If you're running `viam-server` and send it a SIGUSR1 (e.g., you run `pkill -10 viam-server`), it should log stack traces of every active goroutine. and there's a zillion of them that sit around doing mostly nothing (e.g., sleep for 10 seconds, then check if the main config has been updated). but if we got the laggy images and sent a few SIGUSR1's, I wonder if they'd all have a goroutine working inside `rimage.EncodeImage()`.

Reply:
...another way to check on this is to somehow get 2 camera feeds, one using `GetImage` and another using `GetImages`, and see if they differ significantly. but I don't know how easy that would be to do.